R/communities.R

```diff
@@ -55,43 +55,58 @@ leiden.community <- function(graph, resolution=1.0, n.iterations=2) {
 #' @param resolution Integer resoluiton parameter controlling communities detected (default=1.0) Higher resolutions lead to more communities, while lower resolutions lead to fewer communities.
 #' @param niter Number of iterations that the algorithm should be run for (default=2)
 #' @return A vector of membership values
-#' @examples 
+#' @rdname find_partition
+#' @examples
 #' library(igraph)
 #' library(leidenAlg)
-#' 
+#'
 #' g <- make_star(10)
 #' E(g)$weight <- seq(ecount(g))
 #' find_partition(g, E(g)$weight)
-#' 
+#'
 #' @export
 find_partition <- function(graph, edge_weights, resolution=1.0, niter = 2.0) {
+  if (!is(graph, "igraph")) {
+    stop("Input 'graph' must be a valid 'igraph' object")
+  }
+  edgelist <- as.vector(t(igraph::as_edgelist(graph, names=FALSE))) - 1
+  edgelist_length <- length(edgelist)
+  num_vertices <- length(igraph::V(graph)) - 1
+  direction <- igraph::is_weighted(graph)
+  find_partition_rcpp(edgelist, edgelist_length, num_vertices, direction, edge_weights, resolution, niter)
+}
+
+#' @param nrep Number of replicate starts with random number being updated. (default=10) The result with the best quality will be returned.
+#' @rdname find_partition
+#' @export
+find_partition_with_rep <- function(graph, edge_weights, resolution=1.0, niter = 2.0, nrep = 10) {
```

Comment:
@mvfki Please document this as we document the other functions

e.g.

```
#' Finds the optimal partition using the Leiden algorithm
#'
#' @param graph The igraph graph to define the partition on
#' @param edge_weights Vector of edge weights. In weighted graphs, a real number is assigned to each (directed or undirected) edge. For an unweighted graph, this is set to 1. Refer to igraph, weighted graphs.
#' @param resolution Integer resoluiton parameter controlling communities detected (default=1.0) Higher resolutions lead to more communities, while lower resolutions lead to fewer communities.
#' @param niter Number of iterations that the algorithm should be run for (default=2)
#' @return A vector of membership values
#' @examples 
#' library(igraph)
#' library(leidenAlg)
#' 
#' g <- make_star(10)
#' E(g)$weight <- seq(ecount(g))
#' find_partition(g, E(g)$weight)
#' 
#' @export
```

This is for CRAN

Reply:
The tag `@rdname` has already done the job that directs multiple functions with the same "rdname" to one manual where all of them share the argument documentation. Like what we often see with some generic functions, when we do either `?find_partition` or `?find_partition_with_rep`, we get the identical help page.

Reply:
Yes, but
(A) the `nrep` argument is different
(B) I think the function itself is different, and you could provide an explanation to the user explaining this

Reply:
Got you. Will do so, together with the unit tests.